### PR TITLE
Show internal stats for the exporting engine

### LINKED ISSUE #7171: Collect and display internal statistics for Exporting engine
- Respect locks.
- Is it better to print stats from the main thread or from every connector worker?

### PR DESCRIPTION
**Pull request by vlvkobal**
##### Summary
Internal stats for the exporting engine are collected and displayed.

Closes #7171

##### Component Name
exporting engine

##### Test Plan
Check internal statistics in "Netdta Monitoring" menu for every connector and for the main exporting engine thread
image

### CODE REVIEW

File: exporting/send_internal_metrics.c [hunk 1]
```diff
@@ -3,16 +3,170 @@
...
+ * Send the main exporting thread CPU usage
+ *
+ * @param st_rusage a thead CPU usage chart
+ * @param rd_user a dimension for user CPU usage
+ * @param rd_system a dimension for system CPU usage
+ */
+void send_main_rusage(RRDSET *st_rusage, RRDDIM *rd_user, RRDDIM *rd_system)
+{
+    struct rusage thread;
+    getrusage(RUSAGE_THREAD, &thread);
```
Comment (thiagoftsm) [inline exporting/send_internal_metrics.c:35]: This function returns an integer, should we not test and report possible errors?

Comment (vlvkobal) [reply exporting/send_internal_metrics.c:35]: I did it intentionally - it's the scope of the next planned issue #7173.

Comment (vlvkobal) [reply exporting/send_internal_metrics.c:35]: Only two errors are possible in general, but not in this case (as well as in the other 31 places in the Netdata code). So, it doesn't make much sense to do an additional check.

File: exporting/send_internal_metrics.c [hunk 2]
```diff
@@ -3,16 +3,170 @@
...
+
+    // ------------------------------------------------------------------------
+
+    struct rusage thread;
+    getrusage(RUSAGE_THREAD, &thread);
+
+    if (likely(stats->st_rusage->counter_done))
+        rrdset_next(stats->st_rusage);
+
+    rrddim_set_by_pointer(stats->st_rusage, stats->rd_user,   thread.ru_utime.tv_sec * 1000000ULL + thread.ru_utime.tv_usec);
```
Comment (thiagoftsm) [inline exporting/send_internal_metrics.c:168]: Here you are multiplying the seconds by `1000000ULL` and you sum the result to microseconds, please, can you confirm that you did not want multiply the microseconds by 10 seconds?

File: exporting/send_internal_metrics.c (hunk 1 above)
Comment (thiagoftsm) [reply exporting/send_internal_metrics.c:35]: All right right, thank you!

File: exporting/send_internal_metrics.c (hunk 2 above)
Comment (vlvkobal) [reply exporting/send_internal_metrics.c:168]: No. The result in microseconds and is divided by 1000 to get microseconds in the chart.

**Comment by ilyam8** [comment]
@vlvkobal since we have exporting stats how about to add some alarms?

**Comment by vlvkobal** [comment]
It's not a critical thing. I'll add such alarms to our wish list.

**Pull request merged**